components/Entity.js

```diff
@@ -0,0 +1,35 @@
+import React, { PropTypes } from 'react'
+import EntityName from './EntityName'
+import ReferenceEntryList from './ReferenceEntryList'
+
+const Entity = ({ entityId, entity, onEntityNameChange, onAddReferenceEntryClick, onReferenceValueChange, onSynonymsChange }) => (
+  <div>
+    <EntityName entityId={entityId} 
+      name={entity.name}
+      onEntityNameChange={onEntityNameChange} />
+    <ReferenceEntryList entityId={entityId} 
+      entries={entity.entries}
+      onReferenceValueChange={onReferenceValueChange}
+      onSynonymsChange={onSynonymsChange}/>
```

Comment:
There is another way to prevent from passing entityId down and down.
don't pass entityId down here, set like that: 
onReferenceValueChange={t=> onReferenceValueChange(t, entityId) }
onSynonymsChange ={t=> onSynonymsChange(t, entityId) }
it means onReferenceValueChange for Entity is different from for ReferenceEntryList and other components
and by the same way, ReferenceEntry don't pass refId down. all ids for parent are collected at parent.

I'm not sure which one is better but I prefer my way haha
divide and conquer but not let child handle everything